### PR TITLE
WIP: Add stub / brainstorms for the cKDTree module.

### PR DESCRIPTION
**Pull request by rainwoodman**
I am starting to write down the things that I can think of. 

@sturlamolden I'll add you to the repo such that you can directly push into the PR.

### CODE REVIEW

File: scipy-1.0/ckdtree.tex [hunk 1]
```diff
@@ -0,0 +1,35 @@
+The cKDTree module is rewritten in C++ with templated classes.
+
+The support to a box of periodic boundary condition is added. A periodic
```
Comment (tylerjereddy) [inline scipy-1.0/ckdtree.tex]: `to` -> `of` ; `condition` -> `conditions`

File: scipy-1.0/ckdtree.tex [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+The API for querying neighbours is enhanced, to allow querying a subset of
+k nearest neighbours.
+
+\subsubsection{sparse distance matrix}
+
+generating approximated sparse distance matrix
+
+\subsubsection{Pair-Counting}
+
+The cKDTree module implements the dual tree counting algorithm in (cite).
```
Comment (tylerjereddy) [inline scipy-1.0/ckdtree.tex]: The docstring cites `Maneewongvatana and Mount 1999`, which is perhaps: https://arxiv.org/abs/cs/9901013

Conversely, the only reference I see to `dual-tree` in the source code is for `Gray and Moore` : https://arxiv.org/abs/astro-ph/0012333

**Comment by tylerjereddy** [comment]
1 month of stagnation -- merging in the draft to keep things moving forward.

**Comment by rainwoodman** [comment]
Sorry I missed the email notification on your comments.  Maneewongvatana and Mount is relevant on the node split algorithm. Gray and Moore are the dual tree algorithm indeed. The KDTree itself we can probably cite any text book; or we can ask the author of the original cKDTree, @aarchiba.

**Pull request merged**